tests/test_group.rs

```diff
@@ -15,94 +16,146 @@ fn create_commit_optional_path() {
+    let (mls_plaintext_commit, _welcome_bundle_alice_bob_option) = match group_alice_1234
+        .create_commit(
+            group_aad,
+            &alice_credential_bundle,
+            epoch_proposals,
+            true, /* force self-update */
+        ) {
+        Ok(c) => c,
+        Err(e) => panic!("Error creating commit: {:?}", e),
+    };
+    let commit = match &mls_plaintext_commit.content {
+        MLSPlaintextContentType::Commit((commit, _)) => commit,
+        _ => panic!(),
+    };
+    assert!(commit.path.is_some());
 
-    // Alice updates
-    let alice_update_proposal = group_alice_1234.create_update_proposal(
+    // Alice adds Bob without forced self-update
+    // Since there are on Add Proposals, this not generate a path field on the Commit
+    let bob_add_proposal = group_alice_1234.create_add_proposal(
         group_aad,
         &alice_credential_bundle,
-        alice_update_key_package.clone(),
+        bob_key_package.clone(),
     );
-
-    // Only AddProposals
-    let (commit_mls_plaintext, _welcome_option) = match group_alice_1234.create_commit(
-        group_aad,
-        &alice_credential_bundle,
-        vec![bob_add_proposal.clone()],
-        false,
-    ) {
+    let epoch_proposals = vec![bob_add_proposal];
+    let (mls_plaintext_commit, welcome_bundle_alice_bob_option) = match group_alice_1234
+        .create_commit(
+            group_aad,
+            &alice_credential_bundle,
+            epoch_proposals.clone(),
+            false, /* don't force selfupdate */
+        ) {
         Ok(c) => c,
         Err(e) => panic!("Error creating commit: {:?}", e),
     };
-    let (commit, _confirmation_tag) = match commit_mls_plaintext.content {
-        MLSPlaintextContentType::Commit((commit, confirmation_tag)) => (commit, confirmation_tag),
+    let commit = match &mls_plaintext_commit.content {
+        MLSPlaintextContentType::Commit((commit, _)) => commit,
         _ => panic!(),
     };
     assert!(commit.path.is_none());
 
-    // Only AddProposals with forced self update
-    let (commit_mls_plaintext, _welcome_option) = match group_alice_1234.create_commit(
-        group_aad,
-        &alice_credential_bundle,
-        vec![bob_add_proposal],
-        true,
-    ) {
-        Ok(c) => c,
-        Err(e) => panic!("Error creating commit: {:?}", e),
+    // Alice applies Commit
+    match group_alice_1234.apply_commit(mls_plaintext_commit, epoch_proposals, vec![]) {
+        Ok(_) => {}
+        Err(e) => panic!("Error applying commit: {:?}", e),
     };
+    let ratchet_tree = group_alice_1234.tree().get_public_key_tree();
 
-    let (commit, _confirmation_tag) = match commit_mls_plaintext.content {
-        MLSPlaintextContentType::Commit((commit, confirmation_tag)) => (commit, confirmation_tag),
-        _ => panic!(),
+    // Bob creates group from Welcome
+    let mut group_bob_1234 = match MlsGroup::new_from_welcome(
+        welcome_bundle_alice_bob_option.unwrap(),
```

Comment:
Shouldn't this `unwrap` be a test case earlier?

Reply:
Do you mean we should test if the welcome message exists? I can add that test case.

Reply:
Yeah, we shouldn't just panic here when it's missing but check for this earlier.

Reply:
The check is now in L276